include/itkKrcahEigenToScalarImageFilter.hxx

```diff
@@ -41,11 +41,16 @@ KrcahEigenToScalarImageFilter< TInputImage, TOutputImage, TMaskImage >
 ::GenerateInputRequestedRegion()
 {
   Superclass::GenerateInputRequestedRegion();
+
   if ( this->GetInput() )
   {
-  typename TInputImage::Pointer image =
-    const_cast< TInputImage * >( this->GetInput() );
-  image->SetRequestedRegionToLargestPossibleRegion();
+    InputImagePointer image = const_cast< typename Superclass::InputImageType * >( this->GetInput() );
+    image->SetRequestedRegionToLargestPossibleRegion();
```

Comment:
This will limit streaming of the input image and mask, which is fine for now. In the future, if we need to process very large images on systems with limited memory, we can work to set the `RequestedRegion`'s to only the regions required.

Reply:
Yes, good point. Here, parameter estimation takes the 'maximum over all voxels'. Is there a good way to do such an operation with streaming in itk?

Reply:
In practice, we may need to make the maximum a separate input so it can be computed in a streamed way outside of the filter.